packages/core/src/State.ts

```diff
@@ -297,17 +299,22 @@ export class State<
    * Whether the current state value is a subset of the given parent state value.
    * @param parentStateValue
    */
-  public matches<TSV extends TTypestate['value']>(
-    parentStateValue: TSV
+  public matches<TSV extends TTypestate['value'] = never>(
+    parentStateValue: TMeta extends TypegenIsActive
+      ? TMeta['matchesStates']
+      : TSV
   ): this is State<
-    (TTypestate extends any
+    (TMeta extends TypegenIsActive
+      ? never
```

Comment:
could you explain this part? won't this result in `TContext` to be given as `never` to the resulting `State`?

Reply:
Apologies, left this in accidentally.